File: packages/shopify/src/merchantos_shopify/shop_domain.py
import re

from merchantos_domain import InvalidShopDomainError

from merchantos_shopify.constants import SHOP_DOMAIN_PATTERN

_SHOP_RE = re.compile(SHOP_DOMAIN_PATTERN)


def normalize_shop_domain(raw: str) -> str:
    """Accept only `{store}.myshopify.com` (official standalone OAuth regex)."""
    shop = raw.strip().lower()
    shop = shop.removeprefix("https://").removeprefix("http://")
    shop = shop.split("/", 1)[0]
    if not _SHOP_RE.fullmatch(shop):
        raise InvalidShopDomainError("shop must be a *.myshopify.com domain")
    return shop

File: apps/web/lib/shop-domain.ts
const SHOP = /^[a-z0-9][a-z0-9-]*\.myshopify\.com$/i;

export function normalizeShopDomain(raw: string): string {
  return raw.trim().toLowerCase().replace(/^https?:\/\//, "").replace(/\/.*$/, "");
}

export function isValidShopDomain(raw: string): boolean {
  return SHOP.test(normalizeShopDomain(raw));
}
